forge-game/src/main/java/forge/game/ability/effects/DrainManaEffect.java

```diff
@@ -23,13 +30,27 @@ protected String getStackDescription(SpellAbility sa) {
 
     @Override
     public void resolve(SpellAbility sa) {
-        List<Mana> drained = new ArrayList<>();
+        final Game game = sa.getHostCard().getGame();
+        final List<Mana> drained = new ArrayList<>();
+        final Map<Player, Integer> lossMap = Maps.newHashMap();
 
         for (final Player p : getTargetPlayers(sa)) {
             if (!p.isInGame()) {
                 continue;
             }
-            drained.addAll(p.getManaPool().clearPool(false));
+            List<Mana> cleared = p.getManaPool().clearPool(false);
+            drained.addAll(cleared);
+            if (game.getStaticEffects().getGlobalRuleChange(GlobalRuleChange.manaBurn)) {
```

Comment:
use ManaPool.hasBurn instead

or better, if possible remove the GobalRuleChange